components/camera/fake/camera.go

```diff
@@ -170,59 +168,60 @@ func fakeModel(width, height int) (*transform.PinholeCameraModel, int, int) {
-func (c *Camera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
-	img, err := c.getColorImage()
-	if err != nil {
-		return nil, err
+	if c.cacheImage != nil {
+		return c.cacheImage, func() {}, nil
 	}
-	dm, err := c.getDepthImage(ctx)
-	if err != nil {
-		return nil, err
+	width := float64(c.Width)
+	height := float64(c.Height)
+	img := image.NewRGBA(image.Rect(0, 0, c.Width, c.Height))
+
+	totalDist := math.Sqrt(math.Pow(0-width, 2) + math.Pow(0-height, 2))
+
+	var x, y float64
+	for x = 0; x < width; x++ {
+		for y = 0; y < height; y++ {
+			dist := math.Sqrt(math.Pow(0-x, 2) + math.Pow(0-y, 2))
+			dist /= totalDist
+			thisColor := color.RGBA{uint8(255 - (255 * dist)), uint8(255 - (255 * dist)), uint8(0 + (255 * dist)), 255}
+			img.Set(int(x), int(y), thisColor)
+		}
 	}
-	return c.Model.RGBDToPointCloud(img, dm)
-}
 
-// getColorImage always returns the same color image of a chess board.
-func (c *Camera) getColorImage() (*rimage.Image, error) {
-	img, err := rimage.NewImageFromFile(artifact.MustPath("rimage/board2.png"))
-	if err != nil {
-		return nil, err
-	}
-	if c.Height == initialHeight && c.Width == initialWidth {
-		return img, nil
-	}
-	dst := image.NewRGBA(image.Rect(0, 0, c.Width, c.Height))
-	draw.NearestNeighbor.Scale(dst, dst.Bounds(), img, img.Bounds(), draw.Over, nil)
-	return rimage.ConvertImage(dst), nil
+	return rimage.ConvertImage(img), func() {}, nil
 }
 
-// getDepthImage always returns the same depth image of a chess board.
-func (c *Camera) getDepthImage(ctx context.Context) (*rimage.DepthMap, error) {
-	dm, err := rimage.NewDepthMapFromFile(ctx, artifact.MustPath("rimage/board2_gray.png"))
-	if err != nil {
-		return nil, err
-	}
-	if c.Height == initialHeight && c.Width == initialWidth {
-		return dm, nil
+// NextPointCloud always returns a pointcloud of a yellow to blue gradient, with the depth determined by the intensity of blue.
+func (c *Camera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
+	if c.cachePointCloud != nil {
+		return *c.cachePointCloud, nil
 	}
-	dm2, err := rimage.ConvertImageToGray16(dm)
-	if err != nil {
-		return nil, err
+	dm := pointcloud.New()
+	width := float64(c.Width)
+	height := float64(c.Height)
+
+	totalDist := math.Sqrt(math.Pow(0-width, 2) + math.Pow(0-height, 2))
+
+	var x, y float64
+	for x = 0; x < width; x++ {
+		for y = 0; y < height; y++ {
+			dist := math.Sqrt(math.Pow(0-x, 2) + math.Pow(0-y, 2))
+			dist /= totalDist
+			thisColor := color.NRGBA{uint8(255 - (255 * dist)), uint8(255 - (255 * dist)), uint8(0 + (255 * dist)), 255}
+			err := dm.Set(r3.Vector{X: x, Y: y, Z: 255 * dist}, pointcloud.NewColoredData(thisColor))
+			if err != nil {
+				return nil, err
+			}
+		}
 	}
-	dst := image.NewGray16(image.Rect(0, 0, c.Width, c.Height))
-	draw.NearestNeighbor.Scale(dst, dst.Bounds(), dm2, dm2.Bounds(), draw.Over, nil)
-	return rimage.ConvertImageToDepthMap(ctx, dst)
+	return dm, nil
```

Comment:
same, I think you wanna set c.cachePointCloud = dm after the first time you build it